### ISSUE TITLE
Disk shrink is not supported!

### ISSUE THREAD

**Issue by bitchecker** [bpg/terraform-provider-proxmox#1351]
**Describe the bug**
Hi, 
after latest provider update, I was running code in order to find some regression, and it seems that something is changed on the code behavior.
When I try to create a VM template, I download the file from the network and I create a `proxmox_virtual_environment_vm` setting `template  = true`.

When the download finish and it starts to create the VM, the operation fails with error:
```
╷
│ Error: resizing disk: error waiting for VM disk resize: All attempts fail:
│ #1: task "UPID:pve:002B6056:0E86CE6F:665E06F1:resize:101:root@pam:" failed to complete with exit code: shrinking disks is not supported
│ 
```

But the code is the same that previously created the same template that now was destroyed and re-downloaded and created.

I tried also to set up a static disk size with 10GB, but it seems that is always a "shrink" operation. On previous version (sorry but I don't remember which one) I never set up disk size on template and the operations were always fine.

PS: the original qcow2 file is less than 600MB.

**To Reproduce**

1. use `proxmox_virtual_environment_download_file` to download a cloud image from the network
2. use `proxmox_virtual_environment_vm` to create a VM template setting `template  = true`
3. run terraform apply
4. See error `resizing disk: error waiting for VM disk resize: All attempts fail: [...] failed to complete with exit code: shrinking disks is not supported `

**Expected behavior**
The same approach used in previous versions when set up the default disk size for a VM template was optional and the creation was running without issues.

**Additional context**
I tried to set up static 20GB of disk for template, and it seems to work properly but before the updates, it was creating template with a default 8GB.

- Single or clustered Proxmox: single
- Proxmox version: 8.2.2
- Provider version (ideally it should be the latest version): latest
- Terraform/OpenTofu version: v1.8.4
- OS (where you run Terraform/OpenTofu from): Fedora 40

**Comment by bitchecker** [comment]
I found in my git logs, the version of the previous tests!
The provider version was `0.55.1`

**Comment by feynmanliang** [comment]
+1, ran into this when mounting a ZFS hard disk

**Comment by bpg** [comment]
Hi @bitchecker 👋🏼 

The `qcow2` image has a compressed filesystem, so its actual uncompressed size is greater than the allocation you have for the disk, you'll get this error. You can use `qemu-img info` to inspect the image.
A similar use case was discussed here: https://forum.proxmox.com/threads/shrinking-disks-is-not-supported.146842/#post-663270

**Comment by bpg** [comment]
@feynmanliang is your use case the same? Mounting a qcow2 image on a ZFS-based storage?

**Comment by feynmanliang** [comment]
@bpg you're absolutely right, this is user error due to my misunderstanding of qcow2 sparsity.

**Comment by bpg** [comment]
Hey @bitchecker, I'm going to close this issue. Feel free to reopen if you need more help investigating this.

**Comment by bitchecker** [comment]
Hi,
so we're talking about that a cloud image, doubled its size in some month? Ok can be!

My point is: disk size is _Optional_ but has a default of 8GB...can this be updated to the size of the downloaded/imported file? 
Same stuff for classic vm, can we add a check that if the size of the disk is lower than the template size or not set, the size is calculated from the used template?

**Comment by bpg** [comment]
Yeah, that's a good idea!
